segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/OperationProcessor.java

```diff
@@ -112,7 +110,7 @@ protected Duration getShutdownTimeout() {
     protected CompletableFuture<Void> doRun() {
         return Futures
                 .loop(this::isRunning,
-                        () -> delayIfNecessary()
+                        () -> throttle()
```

Comment:
I do not think it is correct to throttle all the operations. Ideally we should only throttle the ops which cause changes to the cache. One simple example is: read from the data which is already in the cache need not be blocked.

Reply:
The Operation Processor does not handle reads. Data that is inserted into the cache via reads is evictable so it does not put pressure on the system in any way. The point of this is to slow down incoming operations in case Tier2 sync falls behind. We cannot accumulate data forever in our Cache.

Also we cannot just throttle those operations that hit the cache since then we would not be honoring our ordering guarantees. Suppose, for example, we have an Append to a segment then a Seal to the same segment. The Seal does not affect the cache, but the Append does. We cannot postpone the append until after the Seal as that would break our ordering guarantees (and cause the append to fail due to the segment being sealed).

Reply:
ok. This arguments makes sense.